wallet/src/receiver.rs

```diff
@@ -65,25 +74,18 @@ pub struct WalletReceiver {
 
 impl Handler for WalletReceiver {
 	fn handle(&self, req: &mut Request) -> IronResult<Response> {
-		let receive: WalletReceiveRequest = serde_json::from_reader(req.body.by_ref())
-			.map_err(|e| IronError::new(e, status::BadRequest))?;
-
-		match receive {
-			WalletReceiveRequest::PartialTransaction(partial_tx_str) => {
-				debug!(LOGGER, "Receive with transaction {}", &partial_tx_str,);
-				receive_json_tx(&self.config, &self.keychain, &partial_tx_str)
-					.map_err(|e| {
-						api::Error::Internal(
-							format!("Error processing partial transaction: {:?}", e),
-						)
-					})
-					.unwrap();
-
-				Ok(Response::with(status::Ok))
-			}
-			_ => Ok(Response::with(
-				(status::BadRequest, format!("Incorrect request data.")),
-			)),
+		let struct_body = req.get::<bodyparser::Struct<JSONPartialTx>>();
```

Comment:
Use bodyparser to turn the request body directly into a JSONPartialTx (and avoid passing the json string representation around).